src/services/payment.service.ts

```diff
@@ -406,16 +447,52 @@ export class PaymentService {
       return {
         clientSecret: '',
         id: '',
+        type: 'fiat',
         invoiceStatus: finalizedInvoice.status,
       };
     }
 
+    const isLifetime = invoiceItem.price?.type === 'one_time';
+
+    if (isLifetime && this.bit2MeService.isAllowedCurrency(currency)) {
+      const invoice = await this.bit2MeService.createCryptoInvoice({
+        description: `Payment for lifetime product ${priceId}`,
+        priceAmount: invoiceItem.amount,
+        priceCurrency: currency,
+        title: `Invoice from Stripe ${finalizedInvoice.id}`,
+        securityToken: jwt.sign(
+          {
+            stripeInvoiceId: finalizedInvoice.id,
+            customerId: customerId,
+          },
+          config.JWT_SECRET,
+        ),
+        foreignId: finalizedInvoice.id,
+        cancelUrl: 'https://drive.internxt.com/cancel',
+        successUrl: 'https://drive.internxt.com/success',
+        purchaserEmail: userEmail || 'hello@internxt.com',
```

Comment:
If this is mandatory, then it should not have a default value. If not, it's pretty strange, unless there is an underlying reason I do not know